src/controllers/main/main.ts

```diff
@@ -1010,19 +1010,34 @@ export class MainController extends EventEmitter {
     })
   }
 
-  async #ensureAccountInfo(accountAddr: AccountId, networkId: NetworkId) {
+  async #ensureAccountInfo(
+    accountAddr: AccountId,
+    networkId: NetworkId
+  ): Promise<{ hasAccountInfo: true } | { hasAccountInfo: false; errorMessage: string }> {
     await this.#initialLoadPromise
     // Initial sanity check: does this account even exist?
     if (!this.accounts.accounts.find((x) => x.addr === accountAddr)) {
-      this.signAccOpInitError = `Account ${accountAddr} does not exist`
-      return
+      return {
+        hasAccountInfo: false,
+        errorMessage: `Account ${accountAddr} does not exist`
+      }
     }
     // If this still didn't work, re-load
     if (!this.accounts.accountStates[accountAddr]?.[networkId])
       await this.accounts.updateAccountState(accountAddr, 'pending', [networkId])
     // If this still didn't work, throw error: this prob means that we're calling for a non-existent acc/network
-    if (!this.accounts.accountStates[accountAddr]?.[networkId])
-      this.signAccOpInitError = `Failed to retrieve account info for ${networkId}, because of one of the following reasons: 1) network doesn't exist, 2) RPC is down for this network`
```

Comment:
Perhaps we should remove `this.signAccOpInitError` entirely? It's still used in `initSignAccOp` and we know the current implementation is problematic, right?

Reply:
It was problematic, only in the case I fixed. In `initSignAccOp` we have several validations, and if we have an errors it would be written to `this.signAccOpInitError`. Later, we are showing this error on SignAccOp screen.